internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -423,6 +421,404 @@ func Test_ValidateClusterClassVariable(t *testing.T) {
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						Default: &apiextensionsv1.JSON{
+							Raw: []byte(`{"spec":{"replicas": -100}}`),
+						},
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"spec": {
+								Type: "object",
+								Properties: map[string]clusterv1.JSONSchemaProps{
+									"replicas": {
+										Type:    "integer",
+										Minimum: pointer.Int64(1),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "pass if variable is an object with a top level default value valid by the given schema",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						Default: &apiextensionsv1.JSON{
+							Raw: []byte(`{"spec":{"replicas": 100}}`),
+						},
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"spec": {
+								Type: "object",
+								Properties: map[string]clusterv1.JSONSchemaProps{
+									"replicas": {
+										Type:    "integer",
+										Minimum: pointer.Int64(1),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		{
+			name: "pass on variable with an example that is invalid by the given schema",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type:      "string",
+						MaxLength: pointer.Int64(6),
+						Example:   &apiextensionsv1.JSON{Raw: []byte(`"short"`)},
+					},
+				},
+			},
+		},
+		{
+			name: "fail on variable with an example that is invalid by the given schema",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type:      "string",
+						MaxLength: pointer.Int64(6),
+						Example:   &apiextensionsv1.JSON{Raw: []byte(`"veryLongValueIsInvalid"`)},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "pass if variable is an object with an example valid by the given schema",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"spec": {
+								Type: "object",
+								Properties: map[string]clusterv1.JSONSchemaProps{
+									"replicas": {
+										Type:    "integer",
+										Minimum: pointer.Int64(0),
+										Example: &apiextensionsv1.JSON{Raw: []byte(`100`)},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "fail if variable is an object with an example invalidated by the given schema",
```

Comment:
```suggestion
			name: "fail if variable is an object with an example valid by the given schema",
```

Reply:
This is correct - the object is invalid here.

Reply:
Ah mom

Reply:
Wrong line :/ Should be l.793

Reply:
:) Fixed now.